app/models/spree/product_question.rb

```diff
@@ -9,12 +9,20 @@ class Spree::ProductQuestion < ActiveRecord::Base
   default_scope -> { order('spree_product_questions.created_at DESC') }
   scope :visible, -> { where(is_visible: true) }
   scope :answered, -> { joins(:product_answer) }
-  scope :not_answered, -> { where.not(id: self.answered.pluck(:id)) }
+  scope :not_answered, -> { left_outer_joins(:product_answer).where(spree_product_answers: { id: nil }) }
   scope :user_product_questions, ->(user_id) { where('spree_product_questions.user_id = ?', user_id) }
 
   validates :content, presence: true
 
   def answer_for_form
     self.product_answer.present? ? self.product_answer : self.build_product_answer
   end
+
+  def self.for_user(user = nil)
+    if user
+      left_outer_joins(:product_answer).where('spree_product_questions.user_id = :user_id OR spree_product_answers.id IS NOT NULL AND spree_product_questions.is_visible = true', user_id: user.id)
```

Comment:
scope is nestable..
I am not sure why .for_user is only fetching visible product questions, 

only admin update the product questions to is_visible once they answer the question..
and this would not give back the question asked by current user until admin has made that decision to update question to is_visible